spec/consensus/evidence.md

```diff
@@ -0,0 +1,196 @@
+# Evidence
+
+Evidence is an important component of Tendermint's security model. Whilst the core
+consensus protocol provides correctness gaurantees for state machine replication
+that can tolerate up to 1/3 failures, the evidence system looks to detect and
+gossip byzantine faults of greater than 1/3 in power. It is worth noting that
+the evidence system is designed purely to detect possible attacks, gossip them,
+commit them on chain and inform the application running on top of Tendermint.
+Evidence in itself does not punish "bad actors", this is left to the discretion
+of the application. A common form of punishment is slashing where the validators
+that were caught violating the protocol have all or a portion of their voting
+power removed. Evidence, given the assumption that 1/3+ of the network is still
+byzantine, is susceptible to censorship and should therefore be considered added
+security on a "best effort" basis.
+
+This document walks through the various forms of evidence, how they are detected,
+gossiped, verified and committed.
+
+> NOTE: Evidence here is internal to tendermint and should not be confused with
+> application evidence
+
+## Detection
+
+### Equivocation
+
+Equivocation is the most fundamental of byzantine faults. Simply put, to prevent
+replication of state across all nodes, a validator tries to convince some subset
+of nodes to commit one block whilst convincing another subset to commit a
+different block. This is achieved by double voting (hence
+`DuplicateVoteEvidence`). A successful duplicate vote attack requires greater
+than 1/3 voting power and a network partition between the aforementioned
+subsets. This is because in consensus, votes are gossiped around. When a node
+observes two conflicting votes from the same peer, it will use the two votes of
+evidence and begin gossiping this evidence to other nodes.
+
+```go
+type DuplicateVoteEvidence struct {
+    VoteA Vote
+    VoteB Vote
+
+    // and abci specific fields
+}
+```
```

Comment:
I wonder whether it would make sense to put the verification rules right here. If you prefer to keep them in separate section, it would be great if you could put a link here to section `### DuplicateVoteEvidence`. (The same comment applies to light client evidence below)

Reply:
My aim was to structure the document in terms of lifecycle events i.e. detection, validation, committing and application as opposed to by evidence type

Reply:
I understand. Then please just add a link to the verification rules. For me, just the data structures without a mention what constitutes valid evidence seems incomplete information, even at this point.